src/components/common/Button.tsx

```diff
@@ -141,20 +134,18 @@ const Button = ({
   fullWidth,
   disabled,
   loading,
-  href,
-  onClick,
+  ...buttonAttributes
 }: ButtonProps) => {
-  return (
-    <ButtonElement
-      href={href}
-      onClick={onClick}
-      className={button({
-        variant,
-        modifier,
-        size,
-        fullWidth,
-        disabled,
-      })}>
+  const classes = button({
+    variant,
+    modifier,
+    size,
+    fullWidth,
+    disabled,
+  });
+
+  const ButtonChildren = () => (
```

Comment:
Instead of rendering the button children outside in a separate component for which I need to make another interface for all the child props, I actually decided to move it in the component itself.